internal/pkg/cli/svc_deploy.go

```diff
@@ -414,6 +420,13 @@ func (o *deploySvcOpts) stackConfiguration(addonsURL string) (cloudformation.Sta
 	var conf cloudformation.StackConfiguration
 	switch t := mft.(type) {
 	case *manifest.LoadBalancedWebService:
+		if err := o.validateAppVersion(t); err != nil {
+			log.Errorf(`Cannot deploy service %s because the application version is incompatible.
+To upgrade the application, please use the application credentials to run %s first (see https://aws.github.io/copilot-cli/docs/credentials/#application-credentials).
```

Comment:
nit: can we remove the app credential stuff, I think they'd understand that `copilot app *` commands need app credentials
```suggestion
To upgrade the application, please run %s first (see https://aws.github.io/copilot-cli/docs/credentials/#application-credentials).
```